src/winning_check.c

```diff
@@ -0,0 +1,258 @@
+                    break;
+                }
+            }
+            return 1;
+        }
+    }
+    highest_c_cart[0] = 0;
+    return 0;
+}// works
+int fstrength_of_pair(card_t *card, int *power) {
+    int count = 1;
+    int value = 0;//is there 4 cart set in ahnd
+    int pairs[4];
+    for (int i = 0; i < 4; ++i) {
+        pairs[i]=0;
+    }
+
+    int how_many_set = 0;
+    for (int i = 0; i < 6; ++i) {
+        if (card[i].value == card[i + 1].value) {
+            count += 1;
+        } else {
+            if (count >= 2) {
+                if (power[count - 1] == 0) { power[count - 1] = card[i - 1].value; }
+                else {
+                    if (power[count - 2] == 0) { power[count - 2] = card[i - 1].value; }
+                }
+                pairs[how_many_set] = count;
+                how_many_set += 1;
+                if (count >= 4) {
+                    value = 1;
+                }
+
+            }
+            count = 1;
+        }
+    }
+    if (count >= 2) {
+        how_many_set += 1;
+        if (power[count - 1] == 0) { power[count - 1] = card[7 - 1].value; }
+        else {
+            if (power[count - 2] == 0) { power[count - 2] = card[7 - 1].value; }
+        }
+    }
+    if (count == 4) { value = 1; }
+    if (count < 2) { count = 0; }
+    for (int i = 0; i < 3; ++i) {
+        count += pairs[i];
+    }
+    if (count <= 5) {
+        if (count != 4) {
+            if (count<4){return count;}
+            return count+1;
+        } else {
+            if (value != 0) {
+                return 7;
+            } else return 2;
+        }
+    }
+    if (count == 6) {
+        if (value != 0) { return 7; }
+        if (how_many_set == 2) { return 6; }
+        else return 2;
+    }
+    return 0;
+}//
+void buble_sort_for_carts(card_t *carts) {
+    for (int i = 0; i < 6; ++i) {
+        for (int j = 0; j < 6; ++j) {
+            if (carts[j].value < carts[j + 1].value) {
+                card_t par = carts[j + 1];
+                carts[j + 1] = carts[j];
+                carts[j] = par;
+            }
+        }
+    }
+int is_in_sequence(card_t *carts, int *start_of_sequence) {
+    int count;
+    start_of_sequence[0] = carts[0].value;
+    for (int i = 0; i < 3; ++i) {
+        count = 1;
+        for (int j = 0; j < 6; ++j) {
+            if (carts[j].value == carts[j + 1].value + 1) {
+                count += 1;
+                if (count > 4) { return 1; }
+            }
+            if (carts[j].value != carts[j + 1].value + 1 && carts[j].value != carts[j + 1].value) {
+                start_of_sequence[0] = carts[j + 1].value;
+                count = 1;
+            }
+        }
+    }
+    if (count == 4 && carts[6].value == 2 && carts[0].value == 14) { return 1; }// check if ace take part in set
+    return 0;
+}
+
+void winner_check(player_t *players, int numbers_of_player, card_t table_carts[5], int *how_many_players_win, player_t *table_of_winners ) {
```

Comment:
it would be better to pass indeces to winning players not `player_t` structures.